src/validator/index.ts

```diff
@@ -48,29 +43,36 @@ type ValidationResult = {
  * @param obj- path to the file
  * @returns A `ValidationResult` object indicating whether the data is valid and any errors that were found.
  */
-function validateObject<T>(obj: any, schemaName: Schema): ValidationResult {
-  const validate = ajv.getSchema<T>(schemaName);
 
-  // Check missing validator
-  if (!validate) {
+function validateObject(obj: unknown, schemaName: Schema): ValidationResult {
+  const schema = schemaMap[schemaName];
+
+  if (!schema) {
     return { valid: false, errors: { schema: "Schema not found" } };
-  } else if (!validate(obj)) {
-    // Enumerate validation errors
-    const errors: Record<string, string> = {};
-    for (const e of validate.errors || []) {
-      const key = e.params.missingProperty || "other";
-      if (key && e.message) {
-        errors[key] = e.message;
-      }
+  }
+
+  const result = schema.safeParse(obj);
+
+  if (result.success) {
+    return { valid: true, errors: {} };
+  }
+
+  const errors: Record<string, string> = {};
+  const formatted = result.error;
+  for (const [key, value] of Object.entries(formatted)) {
+    if (key === "_errors") continue;
+
+    if (value && typeof value === "object" && "_errors" in value) {
```

Comment:
This still looks wrong, I don't think we should be unsafely casting as a ZodFormattedError and directly accessing `_errors`. Can you check the Zod documentation and please use the typed accessors?